nomos-blend/scheduling/src/release_delayer.rs

```diff
@@ -0,0 +1,221 @@
+    current_round: usize,
+    /// The maximum delay, in terms of rounds, between two consecutive release
+    /// rounds.
+    maximum_release_delay_in_rounds: NonZeroUsize,
+    /// The next scheduled round to release the queued messages.
+    next_release_round: usize,
+    /// The random generator to select rounds for message releasing.
+    pub rng: Rng,
+    /// The messages temporarily queued by the scheduler in between release
+    /// rounds.
+    unreleased_messages: Vec<OutboundMessage>,
+}
+
+impl<Rng> SessionProcessedMessageDelayer<Rng>
+where
+    Rng: rand::Rng,
+{
+    /// Initialize the scheduler with the required details.
+    pub fn new(
+        Settings {
+            maximum_release_delay_in_rounds,
+        }: Settings,
+        mut rng: Rng,
+    ) -> Self {
+        debug!(target: LOG_TARGET, "Creating new release delayer with {maximum_release_delay_in_rounds} maximum delay (in rounds).");
+        let next_release_round =
+            schedule_next_release_round_offset(maximum_release_delay_in_rounds, &mut rng);
+        Self {
+            current_round: 0,
+            maximum_release_delay_in_rounds,
+            next_release_round: next_release_round.get(),
+            rng,
+            unreleased_messages: Vec::new(),
+        }
+    }
+
+    /// Poll the scheduler for a potential new release round, while also
+    /// advancing its internal round counter. Rounds follow 0-based indexing, so
+    /// the very first round in a session is considered to be `0`.
+    ///
+    /// **This function is meant to be externally called at every round, as
+    /// failing to do so might result in missed or delayed release rounds**.
+    pub fn poll_next_round(&mut self) -> Option<Vec<OutboundMessage>> {
+        let current_round = self.current_round;
+        // We can safely saturate, since even if we ever reach the end and there's a
+        // scheduled message, it will be consumed the first time the value is hit.
+        self.current_round = self.current_round.saturating_add(1);
+
+        // If it's time to release messages...
+        if current_round == self.next_release_round {
+            // Reset the next release round.
+            self.next_release_round = current_round
+                .checked_add(
+                    schedule_next_release_round_offset(
+                        self.maximum_release_delay_in_rounds,
+                        &mut self.rng,
+                    )
+                    .get(),
+                )
+                .expect("Overflow when scheduling next release round offset.");
+            // Return the unreleased messages and clear the buffer.
+            let messages = mem::take(&mut self.unreleased_messages);
+            debug!(target: LOG_TARGET, "Round {current_round} is a release round.");
+            return Some(messages);
+        }
+        trace!(target: LOG_TARGET, "Round {current_round} is not a release round.");
+        None
+    }
+}
```

Comment:
I'm getting confused with this methods called *poll* and not returning `Poll`. I think we should rename them so avoid this confusion. We can use `check_next_round_ready` or something like that.